### PR TITLE
feat: improve LitElement identification

### PR DESCRIPTION
**Pull request by jpradelle**
Improve algorithm to identify whether a class is a LitElement or not.

This should help with issue #148

### CODE REVIEW

File: src/util.ts [hunk 1]
```diff
@@ -13,13 +13,39 @@ export type DecoratedNode = ESTree.Node & {
   decorators?: BabelDecorator[];
 };
 
+export type NodeWithParent = ESTree.Node & {
```
Comment (43081j) [inline src/util.ts:16]: eslint exports a type for this already, `Rule.Node`

```ts
import type {Rule} from 'eslint';

function hasLitIdentifier(node: Rule.Node): boolean {
  // ...
}
```

File: src/util.ts [hunk 2]
```diff
@@ -13,13 +13,39 @@ export type DecoratedNode = ESTree.Node & {
...
+
 /**
  * Returns if given node has a lit identifier
- * @param {ESTree.Node} node
+ * @param {NodeWithParent} node
  * @return {boolean}
  */
-function hasLitIdentifier(node: ESTree.Node): boolean {
-  return node.type === 'Identifier' && node.name === 'LitElement';
+function hasLitIdentifier(node: NodeWithParent): boolean {
```
Comment (43081j) [inline src/util.ts:25]: i wouldn't actually modify this as it was already doing the correct thing, checking if a node is a lit identifier

we should instead introduce a new function, `hasCustomElementDecorator` or something. then update `isLitClass` to look like this:

```ts
  if (hasCustomElementDecorator(klass)) {
    return true;
  }
  if (clazz.superClass) {
    return (
      hasLitIdentifier(clazz.superClass) || isLitByExpression(clazz.superClass)
    );
  }
  return hasLitIdentifier(clazz);

**Comment by 43081j** [comment]
sorry it took so long to get to this. i've been away for the past month mostly

i think the idea is good, we just need to shuffle some code around

**Comment by jpradelle** [comment]
I pushed a fix, it's cleaner like that :)
There was in fact no need of checking parent node element.

**Review by 43081j — APPROVED** [review]

**Pull request merged**